google_compute_engine/accounts/accounts_utils.py

```diff
@@ -76,10 +76,16 @@ def _CreateSudoersGroup(self):
         self.logger.warning('Could not create the sudoers group. %s.', str(e))
 
     if not os.path.exists(self.google_sudoers_file):
-      with open(self.google_sudoers_file, 'w') as group:
-        message = '%{0} ALL=(ALL:ALL) NOPASSWD:ALL'.format(
-            self.google_sudoers_group)
-        group.write(message)
+      try:
+        with open(self.google_sudoers_file, 'w') as group:
+          message = '%{0} ALL=(ALL:ALL) NOPASSWD:ALL'.format(
+              self.google_sudoers_group)
+          group.write(message)
+      except IOError as e:
+        self.logger.error(
+            'Could not write sudoers file. %s\n%s' % (
+                self.google_sudoers_file, str(e)))
+        return None
```

Comment:
Setting the permissions in an else: clause changes the logic of the implementation. Previously the permissions on the file were set as desired, whether it was written by this code or already in existence, moving the code to an else: implies that permissions are only set when the file is written by this code.

I suppose it is fair to assume that the file has the proper permissions if it exists.